src/agent/application.hpp

```diff
@@ -119,6 +127,73 @@ class Application : private NonCopyable
      */
     otbrError Run(void);
 
+    /**
+     * Returns the OpenThread controller object the application is using.
+     *
+     * @returns OpenThread controller object.
```

Comment:
The `@returns` tag looks redundant. @jwhui Do we always need it?

Reply:
For the core OpenThread project, we currently rely on the brief and tag to populate the OpenThread reference documentation. Example: https://openthread.io/reference/group/plat-alarm#otplatalarmmicrogetnow

Maybe we could omit the brief and only the "Details" table remains. Thoughts?

/cc/ @Vyrastas

Reply:
> Maybe we could omit the brief and only the "Details" table remains. Thoughts?

I would prefer keeping the brief but omitting the "Details", because the brief is used in the summary section. For example, https://openthread.io/reference/group/plat-alarm#summary

Reply:
If we're adding comments, ideally they should be formatted and tagged appropriately as if we expect to generate docs for them in the future.

From @jwhui 's example (otPlatAlarmMicroGetNow), there's no brief tag, the first sentence is what is taken for the summary, the rest of the description is included in the function detail.  I.e. https://github.com/openthread/openthread/blob/df128246ac9fac5f984e7b7f9a1317321185c97d/include/openthread/platform/alarm-micro.h#L74

So for something like this, I would suggest comments like this:

```
     * Get the OpenThread controller object the application is using.
     *
     * @returns The OpenThread controller object.
```

Reply:
Thanks for the suggestion. Sounds good to me. I will update the comments in this PR accordingly.